packages/use-dataloader/src/constants.ts

```diff
@@ -1,4 +1,9 @@
-export const StatusEnum = {
+import { DataLoaderStatus } from './types'
+
+export const StatusEnum: Record<
+  'ERROR' | 'IDLE' | 'LOADING' | 'SUCCESS',
+  DataLoaderStatus
+> = {
```

Comment:
Why not using actal enum ;)
```suggestion
export enum StatusEnum {
```

then replace all `DataLoaderStatus` with the enum type